### ISSUE TITLE
Include some annotation extensions when importing existing annotations

### ISSUE THREAD

**Issue by vanaukenk** [geneontology/noctua-form#124]
From 2020-11 Noctua jamboree:

Curators would like to be able to also import annotation extensions when using the 'Search annotations' functionality.

Could we start with the extensions that use the relations currently in the form, e.g. 'has input'?

This would save curators some time when making their models.

**Comment by tmushayahama** [comment]
@vanaukenk yes, this is actually implemented but was hidden from the menu for extensions (anything without aspect P, F, C). I assume it the same procedure as searching from MF, BP, CC? I have a switch on each Noctua Form line saying allowSearchDatabase true/false

**Comment by vanaukenk** [comment]
@tmushayahama 
Let's look at this together; I'm not sure I understand your comment above.
Thx.

**Comment by vanaukenk** [comment]
Here's an example from the air-2 gene in WB:

![image](https://user-images.githubusercontent.com/1730534/100011645-082b3a00-2da0-11eb-805c-7f2e40861677.png)

**Comment by vanaukenk** [comment]
For the Molecular Function extensions shown above, we'd want a separate row in the Search Annotations pop-up window for each annotation extension.  

For example, for the 'protein kinase activity' annotations, we'd want two rows in the pop-up window for this row:

![image](https://user-images.githubusercontent.com/1730534/100152832-4cd2d600-2e71-11eb-8431-ed9445ea2bac.png)

Each of the two rows would have a separate 'has input' extension with the respective input, air-2 or icp-1.

We can put the Annotation Extension after the Assigned By field.

**Comment by tmushayahama** [comment]
@vanaukenk do you have more examples of GPs to put to get different types of extensions?

**Comment by vanaukenk** [comment]
@tmushayahama 
I think the most critical ones to allow at this point are:

MF annotations with:

- has input
- has output
- part of
- occurs in

BP annotations with:

- part of
- occurs in
- has input
- has output

CC annotations with:

- part of
- existence overlaps
- existence starts and ends during

Note that for the last two CC extension relations we also need to update the Noctua form.  I'll create a separate ticket for that.

**Comment by vanaukenk** [comment]
Checked this on 2021-12-06 workbenches call:
the information is available; we'll add it to the search pop-up box.
Order of columns will match what curators see in the Annotation Preview

**Comment by vanaukenk** [comment]
@tmushayahama - let's look at this one again on the 2022-02-04 workbenches call.

I can select an existing annotation with a 'has input' extension, but the extension information is not then imported into the form editor, nor do I see it on the Noctua form table or the graph.

For an example, we can look at the protein serine/threonine kinase activity MFs for the C. elegans air-2 gene.

**Comment by vanaukenk** [comment]
@tmushayahama 

I've checked 'has input' and 'part of' extensions for MF and BP in this model on noctua-dev: 
http://noctua-dev.berkeleybop.org/workbench/noctua-form/?model_id=gomodel%3A6202e3b700000086

All looks okay on the form and landing page.

**Comment by tmushayahama** [comment]
@vanaukenk there is a part of extension, I didn't know that. I am curious to see how it works since there is a cardinality of [0 - 1] for bp. DO you have an example gene product with part of extension I can test?

**Comment by vanaukenk** [comment]
Looks like the test model on dev is gone, but checking on production (beta).

What populates correctly:

'has input' for MF annotations
'has input' for BP annotations
'occurs in' for BP annotations
'part of' for BP annotations
'part of' for CC annotations

What populates incorrectly:

'occurs in' CC for MF annotations (populated cell type rather than CC)

What doesn't populate at all:

'part of' for MF annotations (e.g. receptor ligand activity part of Wnt signaling pathway)

http://noctua.geneontology.org/workbench/noctua-form-beta/?model_id=gomodel%3A62183af000000728

We can fix the things not working properly in the next Noctua update.
